pkg/pillar/scripts/onboot.sh

```diff
@@ -115,32 +115,64 @@ if [ ! -d $PERSISTDIR/status ]; then
     mkdir $PERSISTDIR/status
 fi
 
-# Checking for low diskspace at bootup. If used percentage of
-# /persist directory is more than 70% then we will remove the
+# percent_used <dataset name> (without leading '/')
+percent_used() {
+    res=$(zfs list -pH -o available,used "$1")
+    # shellcheck disable=SC2181
+    if [ $? = 0 ]; then
+        # shellcheck disable=SC2086
+        avail=$(echo $res | cut -d\  -f1)
+        # shellcheck disable=SC2086
+        used=$(echo $res | cut -d\  -f2)
+        echo $((100*used/(avail+used)))
+    else
+        df --sync /"$1" |awk '{printf("%d",$5);}'
+    fi
+}
+
+# free_space <dataset name> (without leading '/')
+# return value is truncated to MBytes
+# Note that we use df even for zfs since the "available" property in zfs
+# includes unused space in child datasets
+free_space() {
+    ds="$1"
+    res=$(df --sync -k --output=avail "/$ds" | tail -1)
+    echo $((res/1024))
+}
+
+# Checking for low diskspace at bootup.
+# If there is less than 4Mbytes (MIN_DISKSPACE) then remove the content of the
+# following directories in order until we have that amount of available space
 # following sub directories:
-# /persist/log/*
-# /persist/newlog/appUpload/*
-# /persist/newlog/devUpload/*
-# /persist/newlog/keepSentQueue/*
-# /persist/newlog/failedUpload/*
-diskspace_used=$(df /persist |awk '/\/dev\//{printf("%d",$5);}')
-echo "Used percentage of /persist: $diskspace_used"
-if [ "$diskspace_used" -ge "$DISKSPACE_RECOVERY_LIMIT" ]
+PERSIST_CLEANUPS='log netdump kcrashes memory-monitor eve-info-* pubsub-large patchEnvelopesCache patchEnvelopesUsageCache newlog/keepSentQueue newlog/failedUpload newlog/appUpload newlog/devUpload containerd-system-root vault/downloader vault/verifier agentdebug'
```

Comment:
For the memory monitor, a more specific dir should be used: output. I think I mentioned it I the original comment.
```suggestion
PERSIST_CLEANUPS='log netdump kcrashes memory-monitor/output eve-info-* pubsub-large patchEnvelopesCache patchEnvelopesUsageCache newlog/keepSentQueue newlog/failedUpload newlog/appUpload newlog/devUpload containerd-system-root vault/downloader vault/verifier agentdebug'
```

Reply:
Fixed.